src/latexify/codegen/function_codegen.py

```diff
@@ -252,7 +252,15 @@ def visit_FunctionDef(self, node: ast.FunctionDef) -> str:
         body_strs: list[str] = []
 
         # Assignment statements (if any): x = ...
-        for child in node.body[:-1]:
+        for i, child in enumerate(node.body[:-1]):
+            # Allow docstrings
+            if (
+                i == 0
+                and isinstance(child, ast.Expr)
+                and isinstance(child.value, ast.Constant)
```

Comment:
`Constant` is not available in 3.7 (I noticed it through the CI failure).

I think it'd be good if we have a function, e.g., `ast_utils.is_constant(node)` to check if the node is Constant or not.

Reply:
Oversight on my part. It's [available](https://github.com/python/cpython/blob/v3.7.15/Python/Python-ast.c#L962), but [`ast.Str`](https://github.com/python/cpython/blob/v3.7.15/Python/Python-ast.c#L948) isn't a `ast.Constant` for some reason.

Reply:
Oh yeah, both I and you are partially correct. Some node classes are integrated into `Constant` in 3.8. In the official document:

> Deprecated since version 3.8: Old classes ast.Num, ast.Str, ast.Bytes, ast.NameConstant and ast.Ellipsis are still available, but they will be removed in future Python releases. In the meantime, instantiating them will return an instance of a different class.

https://docs.python.org/3.10/library/ast.html#node-classes